logger/regression.go

```diff
@@ -0,0 +1,93 @@
+package logger
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+)
+
+type (
+	// RegressionLogger logs data associated with unique ids, with optional
+	// meta data.
+	RegressionLogger interface {
+		Log(dataID string, data interface{}, meta map[string]interface{}) error
+	}
+
+	// RegressionLoggerCloser is a closable RegressionLogger, which is typically
+	// used when writing log to file.
+	RegressionLoggerCloser interface {
+		io.Closer
+		RegressionLogger
+	}
+
+	regressionLogger struct {
+		w io.WriteCloser
+	}
+
+	filterType string
+
+	// HeaderEntry contains a Key mapped to a regression log meta data Value.
+	HeaderEntry struct {
+		Key   string
+		Value string
+	}
+)
+
+var filters = marshalFilters(
+	"+qStateCounts",
+	"+qGrandTotalRow",
+	"+qDataPages",
+	"+qPivotDataPages",
+	"+qStackedDataPages",
+	"-qNum",
```

Comment:
So this these seems to be filters for _hypercube_ top level, so at least "qListObject" for _listObject_ top level should be added.  Should the rest "top level" data also be in filter?
"hyperCubeDataPages"
"hyperCubeStackPages"
"hyperCubePivotPages"
"listObjectDataPages"
as all data in these should be considered significant.

Reply:
The filters above match what we have for the SDKExerciser regression logging. Not saying that means it's perfectly correct. We might have to play around with the filters once we have the actual differ to see that they have the intended results.

Reply:
The data the filter in SDKExerciser is working on looks different, and my comments wouldn't apply there, except for the qListObject one.

Reply:
If we know data under these top level keys shall be fully included, I can add this now. 

Then we can fine tune as we have the diff tool.

Reply:
for many objects the actual data will be in e.g. _hyperCubeDataPages_ not _hypercube_ using this structure, this is different from how it works in SDKExerciser which works directly on traffic

Reply:
only objects which gets their data in _GetLayout_ request will have the significant portion of data in _hypercube_. This is a large chunk of the objects though.

Reply:
Okay, we still need both 👍 

And `hyperCubeDataPages/**` shall all be included?

Reply:
Added some unit tests where we can test the intended behavior for real logs.